File: beecrowd/python/beginners/1066.py
'''
1066 - PARES, ÍMPARES, POSITIVOS E NEGATIVOS

Leia 5 valores Inteiros. A seguir mostre quantos valores digitados foram pares, quantos valores digitados foram ímpares, quantos valores digitados foram positivos e quantos valores digitados foram negativos.

Entrada
O arquivo de entrada contém 5 valores inteiros quaisquer.

Saída
Imprima a mensagem conforme o exemplo fornecido, uma mensagem por linha, não esquecendo o final de linha após cada uma.
'''
i = 1
contadorPar = 0
contadorImpar = 0
contadorPlus = 0
contadorMinus = 0

while(i <= 5):
	valor = int(input())

	if((abs(valor) % 2) == 0):
		contadorPar += 1
	else:
		contadorImpar += 1

	if(valor > 0):
		contadorPlus += 1
	elif(valor == 0):
		pass
	else:
		contadorMinus += 1

	i += 1

print(f'{contadorPar} valor(es) par(es)')
print(f'{contadorImpar} valor(es) impar(es)')
print(f'{contadorPlus} valor(es) positivo(s)')
print(f'{contadorMinus} valor(es) negativo(s)')

File: beecrowd/cpp/beginners/1066.cpp
#include <iostream>
#include <vector>

int main() {
    int num1, num2, num3, num4, num5;
    std::cin >> num1 >> num2 >> num3 >> num4 >> num5;
    std::vector<int> nums = {num1, num2, num3, num4, num5};

    int pares = 0;
    int impares = 0;
    int positivos = 0;
    int negativos = 0;
    for (int num : nums) {
        (num % 2 == 0) ? pares++ : impares++;

        if (num > 0) {
            positivos++;
        }

        if (num < 0) {
            negativos++;
        }
    }

    std::printf("%i valor(es) par(es)\n", pares);
    std::printf("%i valor(es) impar(es)\n", impares);
    std::printf("%i valor(es) positivo(s)\n", positivos);
    std::printf("%i valor(es) negativo(s)\n", negativos);
}